AUTH-DESIGN.md

```diff
@@ -0,0 +1,53 @@
+# Authentication, authorization and secure communication
+
+* Status: proposed
+* Date: 2024-04-03
+
+Guthub issue: https://github.com/aenix-io/etcd-operator/issues/76
+
+## Security specification
+
+```
+kind: EtcdCluster
+spec:
+...
+  security:
+    peer:
+      caSecretName: peer-ca-tls-secret
+      tlsSecretName: peer-server-tls-secret
+    clientServer:
+      caSecretName: client-server-ca-tls-secret
+      tlsSecretName: client-server-server-tls-secret
+      auth:
+        tlsSecretName: client-server-client-tls-secret
+...
+```
+
+It is expected that secrets contain sections with specific names: `tls.crt`, `tls.key` for tlsSecret and `ca.crt` for caSecret.
```

Comment:
Secret sections (filenames) will be hardcoded to decrease configuration complexity